operations/boot_orchestration/BOS_Workflows.md

```diff
@@ -1,36 +1,361 @@
+
+    ```bash
+    # cray bos v2 sessions create \
+    --template-name SESSIONTEMPLATE_NAME \
+    --operation Boot
+    ```
+
+1. **Session setup operator**
+
+    The creation of a session causes the session-setup operator to set a desired state on all relevant components.
+    This includes pulling files from S3 to determine boot artifacts like kernel, `initrd`, and root file system. The session setup operator also enables the relevant components at this time.
+
+1. **Phase operator (powering-on)**
+
+    The phase operator will detect the enabled components and assign them a phase.  This involves checking the current state of the node, including communicating with HSM to determine the current power status of the node.
+
+    In this example of booting nodes, the first phase is `powering-on`.  If queried at this point, the nodes will have a status of `power-on-pending`.
+
+1. **Power-on operator**
+
+    The power-on operator will detect nodes with a `power-on-pending` status. The power-on operator first sets the desired boot artifacts in BSS.
+    If configuration is enabled for the node, the power-on operator will also call CFS to set the desired configuration and disable the node.
+    The power-on operator then calls CAPMC to power-on the node.
+    Lastly, the power-on operator will update the state of the node in BOS, including setting the last action. If queried at this point, the nodes will have a status of `power-on-called`.
+
+1. **CAPMC boots nodes**
+
+    CAPMC interfaces directly with the Redfish APIs and powers on the selected nodes.
+
+1. **BSS interacts with the nodes**
+
+    BSS generates iPXE boot scripts based on the image content and boot parameters that have been assigned to a node. Nodes download the iPXE boot script from BSS.
+
+1. **S3 interacts with the nodes**
+
+    Nodes download the boot artifacts. The nodes boot using the boot artifacts pulled from S3.
+
+1. **Phase operator (configuring)**
+
+    The phase operator will continue monitoring the states for each node until HSM reports that the power states are on.  
+    When the power state for a node is on, the phase operator will either set the phase to `configuring` if CFS configuration is required or it will clear the current phase if the node is in it's final state.
+
+1. **CFS applies configuration**
+
+    If needed, CFS runs Ansible on the nodes and applies post-boot configuration \(also called node personalization\).
+
+1. **Phase operator (complete)**
+
+    The phase operator will continue monitoring the states for each node until CFS reports that configuration is complete.
+    The phase operator will clear the current phase now that the node is in it's final state. The phase operator will also disable components at this point.
+
+1. **Session completion operator**
+
+    When all nodes belonging to a session have been disabled, the session is marked complete, and it's final status is saved to the database.
+
+### V2 Reboot Nodes
```

Comment:
To the extent that Reboot is copy-pasta from Boot Nodes all of my previous comments apply and will not be reiterated.